src/test/java/com/linkedin/kmf/services/ConsumeServiceTest.java

```diff
@@ -0,0 +1,203 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.utils.SystemTime;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+/**
+ * Unit Testing for the Consume Service Class.
+ */
+public class ConsumeServiceTest {
+  private static final String BROKER_LIST = "localhost:9092";
+  private static final String ZK_CONNECT = "localhost:2181";
+  private static final String TOPIC = "kafka-monitor-topic-testing";
+  private static final Logger LOG = LoggerFactory.getLogger(ConsumeServiceTest.class);
+  private CommitAvailabilityMetrics _commitAvailabilityMetrics;
+  private static final String TAGS_NAME = "name";
+  private static final String METRIC_GROUP_NAME = "commit-availability-service";
+
+  @Test
+  public void commitAvailabilityTest() throws ExecutionException, InterruptedException {
+    ConsumeService consumeService = consumeService();
+
+    MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
+    List<MetricsReporter> reporters = new ArrayList<>();
+    reporters.add(new JmxReporter(Service.JMX_PREFIX));
+    Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
+    Map<String, String> tags = new HashMap<>();
+    String name = "tagName";
+    tags.put(TAGS_NAME, name);
+    _commitAvailabilityMetrics = new CommitAvailabilityMetrics(metrics, tags);
+    Assert.assertNotNull(_commitAvailabilityMetrics._offsetsCommitted.name());
+    Assert.assertNotNull(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue());
+    Assert.assertEquals(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue(), 0.0);
+
+    /* Should start */
+    consumeService.start();
+    Assert.assertTrue(consumeService.isRunning());
+
+    ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+    ScheduledFuture<?> scheduledFuture = executorService.schedule(new Runnable() {
+      @Override
+      public void run() {
+        Assert.assertNotNull(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue());
+        Assert.assertNotEquals(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue(), 0.0);
+      }
+    }, 4, TimeUnit.SECONDS);
+    Assert.assertNotNull(scheduledFuture);
```

Comment:
This code is unlikely to be executed during the duration of the test since nothing waits for it to complete or checks that the exceptions that would be thrown from assertion failures are not there.  Was the intention something like this:
```java
   scheduledFuture.get();  // This will throw ExecutionException
```

Reply:
modified with `scheduledFuture.get();`